File: Excel/700-799/713/713 Challenge.py
import pandas as pd

path = "700-799/713/713 Split and Extract.xlsx"

input = pd.read_excel(path, usecols="A:B", skiprows=1, nrows=4)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=10).rename(columns=lambda x: x.rstrip('.1'))

input = (input.assign(Group=lambda x: x.index + 1, 
                      Data=lambda x: x['Data'].str.split(', '))
             .explode('Data')
             .assign(Position=lambda x: x.groupby('Group').cumcount() + 1,
                     Value=lambda x: x['Position'] * x['Value']))

result = input[['Data', 'Value']].reset_index(drop=True)
print(result.equals(test))

File: Excel/700-799/713/713 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/700-799/713/713 Split and Extract.xlsx"
input = read_excel(path, range = "A2:B6")
test = read_excel(path, range = "D2:E12")

result = input %>%
  mutate(rn = row_number()) %>%
  separate_rows(Data, sep = ", ") %>%
  mutate(Value = Value * row_number(), .by = rn) %>%
  select(-rn)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE
